packages/react/src/shared/proxy/utilsProxy.ts

```diff
@@ -149,17 +149,15 @@ type DecorateProcedure<
 /**
  * @internal
  */
-export type DecoratedProcedureUtilsRecord<TRouter extends AnyRouter> =
-  OmitNeverKeys<{
-    [TKey in keyof TRouter['_def']['record']]: TRouter['_def']['record'][TKey] extends LegacyV9ProcedureTag
-      ? never
-      : TRouter['_def']['record'][TKey] extends AnyRouter
-      ? DecoratedProcedureUtilsRecord<TRouter['_def']['record'][TKey]>
-      : // utils only apply to queries
-      TRouter['_def']['record'][TKey] extends QueryProcedure<any>
-      ? DecorateProcedure<TRouter, TRouter['_def']['record'][TKey]>
-      : never;
-  }>;
+export type DecoratedProcedureUtilsRecord<TRouter extends AnyRouter> = {
+  [TKey in keyof Filter<
```

Comment:
It's worse than doing it inline. TypeScript defers evaluation of property types. But if we do it inline, intellisense will show all properties even though their type is `never`.

Reply:
> But if we do it inline, intellisense will show all properties even though their type is never.

Could you create a minimal repro case for this one? I could dive into the compiler to see if this can be fixed.

Reply:
[Here's a repro](https://www.typescriptlang.org/play?target=6#code/C4TwDgpgBAksEFsoF4oG8BQUpgE4EsBjCALigDsBXBAIwlwG4MBfJjUSWeBAZxSgBKEQgHtcAEwA8PYAXIBzADRdEAPjYdoAcQjAACgEN84uIh6SAKqd5QIAD3jlxfaz1X9M2ANoWA0hBAofHIoAGsAkQAzKCtuHgBdMlizH38QeK8AIjwiCEz42wcIJz4ABixsKAB+CggAN3oK7CTXVID4plYMDFFyGSh5XUNjV35LUftHZxVeVQAKCvw4lriMAEoyHX0jEzjxuPdkd08oXF1KXBClsygDPgNyEE62Xv7rm1QTgzAwABtSdBNHAEYhkACMtz4r2AigqzFh2BoD2RZBO2ByoKgpUhUGhCKg8IqYgeg1RQIxAIATDi8XDYcwoDwDMB8DxIvgIC44i8RH1gDgdqNUINtiM4nN3jw1mwAPQyqAAciR5GRCqCfCiUE0tQauCgNEo-Jk+F+v0ZAAsRAB3YLyKCUMA4gzAkSQXCgKDm+gQDBgQVxAB0yuRTDllXDUAAelUMEA). We meant inlining the filtering in the value position. We could not do it in the key position because we would lose jump to definition.

Reply:
Yeah, that makes sense - when the values are never. I initially misread the comment and thought that the question was about using a `Filter` alias vs using its content "inline" (but still in the `[TKey in keyof _here_]`